src/simplifiers/LARewriter.cc

```diff
@@ -0,0 +1,111 @@
+//
+// Created by Martin Blicha on 03.02.21.
+//
+
+#include "LARewriter.h"
+
+#include "LIALogic.h"
+#include "OsmtApiException.h"
+
+std::size_t DivModRewriter::counter = 0;
+const char * DivModRewriter::divPrefix = ".div";
+const char * DivModRewriter::modPrefix = ".mod";
+
+DivModRewriter::DivModPair DivModRewriter::freshDivModPair(LIALogic & logic) {
+    std::string num = std::to_string(counter++);
+    std::string divName = divPrefix + num;
+    std::string modName = modPrefix + num;
+    return {logic.mkNumVar(divName.c_str()), logic.mkNumVar(modName.c_str())};
+}
+
+PTRef rewriteDivMod(LIALogic & logic, PTRef root) {
+    return DivModRewriter().rewriteDivMod(logic, root);
+}
+
+// TODO: figure out how to cache assigned variables to avoid duplication
```

Comment:
Is this comment still valid?  It looks like there is a cache already in place, but maybe it refers to something else?

Reply:
I think it is dealt with now.